mpqp/core/instruction/measurement/expectation_value.py

```diff
@@ -194,60 +194,65 @@ class ExpectationMeasure(Measure):
 
     def __init__(
         self,
-        targets: list[int],
         observable: Observable,
+        targets: Optional[list[int]] = None,
         shots: int = 0,
         label: Optional[str] = None,
     ):
         from mpqp.core.circuit import QCircuit
 
-        super().__init__(targets, shots, label)
-        self.observable = observable
-        """See parameter description."""
-        # Raise an error if the number of target qubits does not match the size of the observable.
-        if self.nb_qubits != observable.nb_qubits:
-            raise NumberQubitsError(
-                f"{self.nb_qubits}, the number of target qubit(s) doesn't match"
-                f" {observable.nb_qubits}, the size of the observable"
-            )
+        if targets is None:
```

Comment:
in this case we don't define other attributes like `pre_measure` won't this pose problems at execution time ?

Reply:
done